### ISSUE TITLE
Dots are not allowed in the name for the Server Custom Resource

### ISSUE THREAD

**Issue by SDJustus** [SeldonIO/seldon-core#4818]
The scheduler doesn't notice a Server created with a dot inside it's name:

For example: 
The Server with the name triton-gpu-2301 get deployed from the scheduler as expected. 
If I try to deploy the same server changing only the name to triton-gpu-23.01, the Custom Resource is created, but the Scheduler doesn't deploy (or even notice) the new Server. 

Is this expected behavior? I that is the case, would it be meaningful to not allow a Server creation with a dot inside its name in the first place?

**Comment by cliveseldon** [comment]
Yes, we should limit to acceptable Kubernetes names.

**Comment by SDJustus** [comment]
so this would alphanumeric and "-" only.. Am i right on this?

**Comment by cliveseldon** [comment]
Well I am surprised the Server was allowed as your name ends in a numeric: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

**Comment by SDJustus** [comment]
No error when deploying the name without the dot

**Comment by SDJustus** [comment]
so it seems the Server is not noticed by the Scheduler, because the controller tries to create a Service with a dot inside it's name, which seems to be not allowed

**Comment by cliveseldon** [comment]
Yes Server names need to respect rfc-1035 as they will be used to create Service names:

 https://github.com/kubernetes/kubernetes/blob/0e3818e02760afa8ed0bea74c6973f605ca4683c/pkg/apis/core/validation/validation.go#L248-L251

**Comment by cliveseldon** [comment]
I'll create  a PR to make the error visible in the Status of the Server

**Comment by SDJustus** [comment]
Thank you very much!
